trade-finance-manager-api/src/graphql/reducers/mappings/facilities/mapCoverEndDate.js

```diff
@@ -1,20 +1,37 @@
-const moment = require('moment');
+const { fromUnixTime, format } = require('date-fns');
 const { hasValue } = require('../../../../utils/string');
 const { formatYear } = require('../../../../utils/date');
+const api = require('../../../../v1/api');
+const { latestAmendmentCoverEndDateAccepted } = require('../../../helpers/amendment.helpers');
 
-const mapCoverEndDate = (day, month, year) => {
+const mapCoverEndDate = async (day, month, year, facilitySnapshot) => {
   const hasCoverEndDate = (hasValue(day)
     && hasValue(month)
     && hasValue(year));
 
   if (hasCoverEndDate) {
-    const coverEndDate = moment().set({
-      date: Number(day),
-      month: Number(month) - 1, // months are zero indexed
-      year: formatYear(Number(year)),
-    });
+    let dayToUse = day;
+    let monthToUse = month;
+    let yearToUse = year;
 
-    return moment(coverEndDate).format('D MMMM YYYY');
+    if (facilitySnapshot) {
+      const { _id } = facilitySnapshot;
+
+      const latestCompletedAmendment = await api.getLatestCompletedAmendment(_id);
+
+      if (latestCompletedAmendment?.amendmentId && latestCompletedAmendment?.coverEndDate && latestAmendmentCoverEndDateAccepted(latestCompletedAmendment)) {
```

Comment:
same as the other file, you don't need the `latestCompletedAmendment?.amendmentId` if you are checking `latestCompletedAmendment?.coverEndDate`

Reply:
done